flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/ReconciliationUtils.java

```diff
@@ -255,4 +259,45 @@ public static boolean shouldRollBack(
                 .minus(readinessTimeout)
                 .isAfter(Instant.ofEpochMilli(reconciliationStatus.getReconciliationTimestamp()));
     }
+
+    public static boolean shouldRecoverJmDeployment(
+            FlinkDeploymentStatus status, Configuration conf) {
+        boolean enabled =
+                conf.getOptional(
+                                KubernetesOperatorConfigOptions
+                                        .OPERATOR_RECOVER_JM_DEPLOYMENT_ENABLED)
+                        .orElse(
+                                conf.get(FlinkConfigBuilder.FLINK_VERSION)
+                                                .isNewerVersionThan(FlinkVersion.v1_14)
+                                        ? true
+                                        : false);
+
+        return enabled
```

Comment:
Will this also trigger a redeployment for a batch job finished ?

Reply:
Not in 1.15 as the cluster won't shut down (so the jobmanager deployment is not lost). In flink 1.14 if you enable this it would redeploy the batch job, this is exactly the reason why this is disabled in 1.14 by default.

Reply:
Get it

Reply:
Get it